tests/backends/test_pyevm.py

```diff
@@ -24,4 +23,31 @@ def eth_tester():
 
 
 class TestPyEVMBackendDirect(BaseTestBackendDirect):
-    pass
+
+    def test_override_genesis_parameters(self, eth_tester):
+        super().test_reset_to_genesis(eth_tester)
+
+        param_overrides = {'gas_limit': 4750000}
+
+        # Use the the static method to generate custom genesis params
+        genesis_params = PyEVMBackend.generate_genesis_params(param_overrides)
+        assert genesis_params['block_number'] == 0
+        assert genesis_params['gas_limit'] == 4750000
+
+        # Only default genesis paraeter keys can be overriden
+        invalid_overrides = {'gato': 'con botas'}
+        with pytest.raises(ValueError):
+            _invalid_genesis_params = PyEVMBackend.generate_genesis_params(invalid_overrides)
+
+        # Initialize pyevm backend with genesis params
+        pyevm_backend = PyEVMBackend(genesis_params)
+        assert pyevm_backend.chain.header._gas_limit == 4745362
```

Comment:
How about using a public API instead, in case this private variable gets refactored? Something like:

```py
genesis_block = pyevm_backend.get_block_by_number(0)
assert genesis_block.gas_limit == 4750000
```